src/components/MoneyReportHeader.tsx

```diff
@@ -146,11 +120,13 @@ function MoneyReportHeader({session, personalDetails, policy, chatReport, nextSt
                 {shouldShowSettlementButton && !isSmallScreenWidth && (
                     <View style={styles.pv2}>
                         <SettlementButton
+                            // @ts-expect-error : TODO #25100 : type of currency should be handled in component
                             currency={moneyRequestReport.currency}
                             policyID={moneyRequestReport.policyID}
-                            chatReportID={chatReport.reportID}
+                            chatReportID={chatReport?.reportID}
                             iouReport={moneyRequestReport}
-                            onPress={(paymentType) => IOU.payMoneyRequest(paymentType, chatReport, moneyRequestReport)}
+                            // @ts-expect-error : TODO #24926 : type of currency should be handled in file IOU.js
+                            onPress={(paymentType: DeepValueOf<typeof CONST.IOU.PAYMENT_TYPE>) => IOU.payMoneyRequest(paymentType, chatReport, moneyRequestReport)}
```

Comment:
Let's create a separate type for payment type
```
type PaymentType = DeepValueOf<typeof CONST.IOU.PAYMENT_TYPE>;
```
```suggestion
                            onPress={(paymentType: PaymentType) => IOU.payMoneyRequest(paymentType, chatReport, moneyRequestReport)}
```